File: 1523-count-odd-numbers-in-an-interval-range/1523-count-odd-numbers-in-an-interval-range.py
class Solution:
    def countOdds(self, low: int, high: int) -> int:
        """
        want the count of odd numbers - INCLUDING the two boundaries
        
        o  o
        3, 7
        range = 4
        > 3 odds
        
        e  e
        4, 8
        range = 4
        > !!! only 2 odds between here!
        
        e  o
        4, 9
        range = 5
        > 3 odds
        
        o  e
        3, 8
        range = 5
        > 3 odds
        
        take the floor of the range, add 1?
        unless both nums are even
        
        low  |  high
          o  |  o
          e  |  e
          e  |  o
          o  |  e
        """
        
        if (high % 2 == 0) and (low % 2 == 0):
            return (high - low) // 2
        return ((high - low) // 2) + 1

File: 1523-count-odd-numbers-in-an-interval-range/1523-count-odd-numbers-in-an-interval-range.java
class Solution {
    public int countOdds(int low, int high) {
        /*
            both odd -> (high - low // 2) + 1
            one even -> (high - low // 2) + 1
            both even -> high - low // 2
            
            3, 7 -> 3
            
            3, 8 -> 3
            4, 9 -> 3
            
            10, 16 -> 3 -> hi - low // 2
        */
        int count = (high - low) / 2;
        
        if ((low & 1) == 0 && (high & 1) == 0) return count;
        else return count + 1;
    }
}